cmd/siklu/bank_management.c

```diff
@@ -54,6 +56,27 @@ struct software_bank_t* bank_management_get_current_bank(void) {
 		goto fail_free;
 	}
 	
+	config_boot_tries_left = siklu_fdt_getprop_string(fdt, "/", PROP_BOOT_TRIES_LEFT, NULL);
+	if (IS_ERR(config_boot_tries_left)) {
+		printf("Could not read boot_tries_left\n");
+	} else {
+		printf("boot_tries_left = %s\n", config_boot_tries_left);
+		boot_tries_left = simple_strtol(config_boot_tries_left, NULL, 10);
+	}
+	if (boot_tries_left >= 0) {
+		// decrement boot_tries_left for next boot
+		char boot_tries_left_str[5];
+		if (sprintf(boot_tries_left_str, "%d", boot_tries_left - 1) > 0) {
+			siklu_fdt_setprop_string(fdt, "/", PROP_BOOT_TRIES_LEFT, boot_tries_left_str);
+		}
+	}
+	if (boot_tries_left == 0) {
+		//out of tries, switch bank
+		bank = (bank == &first_bank) ? &second_bank : &first_bank;
```

Comment:
maybe extract this into a function?

Reply:
There were 2 separate functions (read-decrease-save counter and switch-save bank), before it came to writing fdt back. It takes a lot to optimize 2 separate writes based on result of 2 separate functions.

Reply:
done